### PR TITLE
Add -Wall when building system libraries. NFC

### PR DESCRIPTION
**Pull request by sbc100**
For libc I did have to turn off a bunch more wanrings, but I think this is still worth it.

### CODE REVIEW

File: system/lib/libc/musl/src/internal/atomic.h [hunk 1]
```diff
@@ -194,7 +194,7 @@ static inline void a_store(volatile int *p, int v)
 
 #ifndef a_barrier
 #define a_barrier a_barrier
-static void a_barrier()
+static inline void a_barrier()
```
Comment (tlively) [inline system/lib/libc/musl/src/internal/atomic.h:197]: Is this going to be a pain when we update musl?

File: system/lib/libc/musl/src/math/ceil.c [hunk 2]
```diff
@@ -1,11 +1,13 @@
 #include "libm.h"
 
+#ifndef __wasm__
```
Comment (tlively) [inline system/lib/libc/musl/src/math/ceil.c:3]: Why the new ifdefs?

**Review by tlively — APPROVED** [review]
LGTM % questions

File: system/lib/libc/musl/src/math/ceil.c (hunk 2 above)
Comment (sbc100) [reply system/lib/libc/musl/src/math/ceil.c:3]: The stuff inside these block is not used at all in the `__wasm__` case below

**Review by kripken — APPROVED** [review]
Slight worry about the musl changes increasing the effort on upgrades, but they do look useful, so if you think it's worthwhile sgtm.

**Comment by sbc100** [comment]
> Slight worry about the musl changes increasing the effort on upgrades, but they do look useful, so if you think it's worthwhile sgtm.

musl updates have gotten a lot easier now that we maintain our patches in the emscirpten-core/musl repo and use the `push_musl_changes.py` and `update_musl.py` scripts.

File: system/lib/libc/musl/src/internal/atomic.h (hunk 1 above)
Comment (sbc100) [reply system/lib/libc/musl/src/internal/atomic.h:197]: No, musl patches are maintained in git these days and we have scripts for pushing and pulling them,

**Pull request merged**